### PR TITLE
Add BH properties from HYBRID model

### PR DESCRIPTION
**Pull request by robjmcgibbon**
We have two different flavours of COLIBRE - thermal only AGN feedback and hybrid AGN feedback. The hybrid runs output some additional BH properties which we don't have for the thermal model. This PR adds these new properties, and also a couple of extra properties for the thermal runs.

This requires a separate parameter file for the thermal and hybrid runs. We also need to take into account that the COLIBRE snipshots can't calculate all the same properties as the snapshot. Rather than having 4 parameter files, I have added some functionality to the parameter file reader such that we can specify properties as on/off in snipshots. I have added a new runtime option (`--snipshot`) which should be passed to `compute_halo_properties.py` when running on a snipshot.

An additional minor change is to remove the test parameter files, and generate them when running the tests. This is done by the `tests/COLIBRE/create_parameters_file.py` script.

### CODE REVIEW

File: aperture_properties.py [hunk 1]
```diff
@@ -2755,6 +2933,22 @@ class ApertureProperties(HaloProperty):
             "BHmaxvel",
             "BHmaxAR",
             "BHmaxlasteventa",
+            "BlackHolesTotalInjectedEnergy",
```
Comment (FHusko) [inline aperture_properties.py:2936]: One small thing: could you add an equivalent of this quantity, `BlackHolesTotalInjectedJetEnergy`, that is defined as the sum over all injected jet energies from all the BHs?

Comment (robjmcgibbon) [reply aperture_properties.py]: https://github.com/SWIFTSIM/SOAP/pull/97/commits/f77ce2d6be819765a685be2618366cac76ba19e5

**Comment by robjmcgibbon** [comment]
@VictorForouhar can you please approve, Filip doesn't have permission

**Review by VictorForouhar — APPROVED** [review]
Didn't check for correctness. Filip did.

**Pull request merged**